account/middleware.py

```diff
@@ -51,3 +62,32 @@ def process_request(self, request):
             if account:
                 tz = settings.TIME_ZONE if not account.timezone else account.timezone
                 timezone.activate(tz)
+
+
+class ExpiredPasswordMiddleware(object):
+
+    def process_request(self, request):
+        if request.user.is_authenticated() and not request.user.is_staff:
+            next_url = resolve(request.path).url_name
+            # Authenticated users must be allowed to access
+            # "change password" page and "log out" page.
+            # even if password is expired.
+            if next_url not in [settings.ACCOUNT_PASSWORD_CHANGE_REDIRECT_URL,
+                                settings.ACCOUNT_LOGOUT_URL,
+                                ]:
+                if check_password_expired(request.user):
+                    signals.password_expired.send(sender=self, user=request.user)
+                    messages.add_message(
+                        request,
+                        messages.WARNING,
+                        _("Your password has expired. Please save a new password.")
+                    )
+                    redirect_field_name = "next"  # fragile!
```

Comment:
I agree with removing fragility and standardizing that field name.

We might consider relying on / aliasing from `django.contrib.auth.REDIRECT_FIELD_NAME`, but still allowing the the overrides on the existing classes/methods above.

So `REDIRECT_FIELD_NAME` is the default unless overridden.